File: Desenvolvimento/client/src/@fuse/types/models/monitoramento-barragem.ts
import { BaseModel } from '../base-model';
import { Barragem } from './barragem';
import { NivelMonitoramento } from './nivel-monitoramento';
import { UnidadeMedida } from './unidade-medida';
import { Sensor } from './sensor';
import { Consultoria } from './Consultoria';
import { TipoMonitoramento } from "./tipo-monitoramento";

export class MonitoramentoBarragem extends BaseModel {
    constructor(
        public id: number,
        public descricao?: string,
        public observacao?: string,
        
        public barragemId?: number,
        public barragem?: Barragem,
        
        public nivelMonitoramentoId?: number,
        public nivelMonitoramento?: NivelMonitoramento,

        public nivel?: number,
        
        public unidadeMedidaId?: number,
        public unidadeMedida?: UnidadeMedida,
        
        public dataHora?: Date,
        public latitude?: string,
        public longitude?: string,

        public sensorId?: number,
        public sensor?: Sensor,

        public consultoriaId?: number,
        public consultoria?: Consultoria,

        public tipoMonitoramentoId?: number,
        public tipoMonitoramento?: TipoMonitoramento

    ) { super(); }
}

File: Desenvolvimento/server/src/ApplicationCore/Entities/MonitoramentoBarragem.cs
using ApplicationCore.Enums;
using ApplicationCore.Interfaces.Base;
using System;
using System.Collections.Generic;
using System.Text;

namespace ApplicationCore.Entities
{
    public class MonitoramentoBarragem : IBaseEntity<long>
    {
        public long Id { get; set; }

        public long BarragemId { get; set; }
        public virtual Barragem Barragem { get; set; }

        public string Descricao { get; set; }
        public string Observacao { get; set; }

        public long NivelMonitoramentoId { get; set; }
        public virtual NivelMonitoramento NivelMonitoramento { get; set; }

        public float Nivel { get; set; }
        public long? UnidadeMedidaId { get; set; }
        public virtual UnidadeMedida UnidadeMedida { get; set; }

        public DateTime DataHora { get; set; }
        public string Latitude { get; set; }
        public string Longitude { get; set; }

        public long? SensorId { get; set; }
        public virtual Sensor Sensor { get; set; }

        public long? ConsultoriaId { get; set; }
        public virtual Consultoria Consultoria { get; set; }

        public long TipoMonitoramentoId { get; set; }
        public virtual TipoMonitoramento TipoMonitoramento { get; set; }

    }
}
